### PR TITLE
[Merged by Bors] - add caching to test suite

### PR DESCRIPTION
**Pull request by realbigsean**
## Issue Addressed

N/A

## Proposed Changes

Add some caching to the test suite and to the aarch64 cross-compile in the docker build. 

## Additional Info

Cache hits only occur if the Cargo.lock file is unchanged, Github Actions runner OS matches, and the cache is "in scope". Some documentation on github actions cache scoping is here:

https://docs.github.com/en/free-pro-team@latest/actions/guides/caching-dependencies-to-speed-up-workflows#matching-a-cache-key

I'm not sure how frequently we'll get cache hits, I imagine only on smaller PR's or updates to the same PR.  And there is a cache size limit that we may end up reaching quickly.  But Github actions handles evictions if we go over that limit. 

Not sure how much of an impact this will end up having but I don't really see a downside to trying it out.

### CODE REVIEW

**Comment by realbigsean** [comment]
found a downside: caching without specifying `portable` led to `Illegal instruction` errors 😭 

I guess the CPUs these servers are using aren't consistent

**Comment by realbigsean** [comment]
cache hits seem to shave off 10-15 minutes of the total run

**Review by AgeManning — APPROVED** [review]
Looks and sounds good to me. 

Lets give it a shot. 

bors r+

File: .github/workflows/test-suite.yml [hunk 1]
```diff
@@ -3,14 +3,15 @@ name: test-suite
 on:
   push:
     branches:
-      - master
+      - unstable
```
Comment (michaelsproul) [inline .github/workflows/test-suite.yml:6]: IMO this should be `stable` as well, but we can do that in a separate PR
